Jellyfin.Server/Program.cs

```diff
@@ -397,6 +403,12 @@ private static ServerApplicationPaths CreateApplicationPaths(StartupOptions opti
                 }
             }
 
+            // Reset webDir if the directory does not exist, or is empty
+            if (!Directory.Exists(webDir) || !Directory.GetFiles(webDir).Any())
```

Comment:
I'm actually worried by this behaviour of silently ignoring user errors. I would rather we have a separate switch that turns this behaviour on instead of sweeping potential user error under the rug.

Reply:
Why not just log it and move on?

Reply:
Because for an average user missing web directory is a fatal error, it only makes sense in a very special case of someone developing the web ui, in which case having a separate switch doesn't seem too bad.

Reply:
It will also have use for packaging web and server separately with Debuntu packages (the long-term plan for this), but I agree it should be configurable from the CLI, so that this can be explicitly set in the per-distro configuration.

Reply:
@JustAMan Yeah, that makes sense I'll add a config flag to make this behavior explicit. I guess `IsHostingContent` would make sense to match the property in appHost. Should we set the flag by default to `true` or `false`?

My initial motivation for this was to improve the onboarding experience for new developers. Ideally a developer would be able to pull this repo, open it up in VS and just run the project. Specifying an additional flag is a pretty minor additional step, so maybe we really just need to improve the developer documentation. Neither the [Contributing](https://jellyfin.org/docs/general/contributing/index.html) or [Building From Source](https://jellyfin.org/docs/general/administration/building.html) pages actually have instructions on how a developer can get started with the server project, so that should probably be added somewhere and we could document this flag there as well.

Reply:
You shouldn't be easing developers' life at the cost of making the software more error-prone to an unaware user :)

As for the flag, I would say default it to "yes, I host content, so error out if web directory is not there".

Reply:
We don't want to throw an error for new developers that cloned the server and tried to run it either. That's not a good first experience and will deter lots of people who might have made a small improvement.

Reply:
You're caring for devs, I'm caring for users.
Who's going to have more troubles figuring why stuff doesn't work if they are somehow missing a web folder? ;)

Reply:
I think defaulting to "yes, host content" makes sense. Even if we set the default the other way for devs, they'll still launch the project and be greeted with the swagger API page which isn't particularly helpful either unless they are expecting it. What we really need to do for devs is provide instructions for running the server that includes directions for both methods of hosting the web content (statically and with Webpack). For most projects this is features directly in the main README, I can add those instructions there if there are no objections.